base/server/python/pki/server/deployment/__init__.py

```diff
@@ -114,6 +114,21 @@ def init(self, parser):
         self.config_client = util.ConfigClient(self)
         self.parser = parser
 
+    def validate(self):
+        # Validate environmental settings for the deployer;
+        # to be called before self.init().
+
+        blacklisted_hostnames = ['localhost', 'localhost.localdomain',
```

Comment:
This will prevent developers to test it on a local system. I have all my instances installed on `localhost.localdomain` -- Doesn't require users to play around with `/etc/hosts` to test our product. Do we need this strict check? Instead, what about just throwing a warning and showing it in the console (even when `pkispawn` is executed in non-verbose mode)?

@edewata  what do you think?

Reply:
Yeah, how about displaying the actual hostname as information (not a warning or an error) during installation? If the hostname is one of these localhost names, we can display a warning saying that it may not work with systems running on a different host. If the admin doesn't use another system, they can simply ignore the warning.

Reply:
Sure, the issue I have with displaying the hostname is that it would disappear quickly, especially with `-v` flag. Considering it breaks cloning, I'd consider it a bug to have `localhost` as the domain...

How about a flag `--allow-localhost` on `pkispawn` for local development?

Reply:
@SilleBille It won't prevent developers from testing on their local system -- they just need to set a domain name for `127.0.0.1` in `/etc/hosts`. Definitely possible for anyone who has admin access on their machine. 

(as an aside) It is also required for testing non-Dogtag products like `gssproxy`, `python-gssapi`, and `krb5`.

This probably needs a corresponding docs change.